### ISSUE TITLE
UTs fails on Java11

### ISSUE THREAD

**Issue by cstamas** [eclipse-sisu/sisu.plexus#2]
When built on Java11 (Linux), the build fails as UTs fail during build.

```
Failed tests: 
  ComponentAnnotationTest.testComponentImpl:77->checkBehaviour:104 expected:<[, alias="", configurator="", composer="", isolatedRealm=false, role=org.eclipse.sisu.plexus.ComponentAnnotationTest$A.class, type="", instantiationStrategy="", description="", version="", hint="", @org.codehaus.plexus.component.annotations.Component, lifecycleHandler="", profile="", factory=""]> but was:<[, alias=, factory=, version=, composer=, lifecycleHandler=, type=, isolatedRealm=false, configurator=, role=interface, @org.codehaus.plexus.component.annotations.Component, profile=, instantiationStrategy=, description=, hint=, org.eclipse.sisu.plexus.ComponentAnnotationTest$A]>
  RequirementAnnotationTest.testRequirementImpl:50->checkBehaviour:77 expected:<[, role=java.lang.Object.class, hints={}, optional=false, @org.codehaus.plexus.component.annotations.Requirement, hint=""]> but was:<[, role=class, hints=[], optional=false, @org.codehaus.plexus.component.annotations.Requirement, java.lang.Object, hint=]>
  ConfigurationAnnotationTest.testConfigurationImpl:35->checkBehaviour:56 expected:<[name="", , value="Default", @org.codehaus.plexus.component.annotations.Configuration]> but was:<[, value=Default, name=, @org.codehaus.plexus.component.annotations.Configuration]>

Tests run: 88, Failures: 3, Errors: 0, Skipped: 0
```

Plaf:
```
cstamas@Urnebes ~/Worx/eclipse/sisu.plexus  (master)$ uname -a
Linux Urnebes 5.4.0-84-generic #94-Ubuntu SMP Thu Aug 26 20:27:37 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
cstamas@Urnebes ~/Worx/eclipse/sisu.plexus  (master)$ java -version
openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment AdoptOpenJDK-11.0.11+9 (build 11.0.11+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK-11.0.11+9 (build 11.0.11+9, mixed mode)
cstamas@Urnebes ~/Worx/eclipse/sisu.plexus  (master)$ mvn -v
Apache Maven 3.8.2 (ea98e05a04480131370aa0c110b8c54cf726c06f)
Maven home: /home/cstamas/.sdkman/candidates/maven/current
Java version: 11.0.11, vendor: AdoptOpenJDK, runtime: /usr/lib/jvm/adoptopenjdk-11-hotspot-amd64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.4.0-84-generic", arch: "amd64", family: "unix"
cstamas@Urnebes ~/Worx/eclipse/sisu.plexus  (master)$ 
```

**Comment by mcculls** [comment]
these tests check the `toString` output of the annotation implementations have the same content as the JDK's output - Java11 changed the ordering for `Annotation.toString` so we need to make this check less strict to account for this.

**Comment by mcculls** [comment]
Fixed by d24bfa2
